crates/client/db/tests/test_block.rs

```diff
@@ -0,0 +1,295 @@
+
+    assert_eq!(backend.resolve_block_id(&BlockId::Hash(block_hash)).unwrap().unwrap(), DbBlockId::BlockN(0));
+    assert_eq!(backend.resolve_block_id(&BlockId::Number(0)).unwrap().unwrap(), DbBlockId::BlockN(0));
+    assert_eq!(backend.resolve_block_id(&DbBlockId::Pending).unwrap().unwrap(), DbBlockId::Pending);
+}
+
+#[tokio::test]
+async fn test_block_id_not_found() {
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    assert!(backend.resolve_block_id(&BlockId::Hash(Felt::from(0))).unwrap().is_none());
+}
+
+#[tokio::test]
+async fn test_store_block() {
+    const BLOCK_ID_0: DbBlockId = DbBlockId::BlockN(0);
+
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    assert!(backend.get_block(&BLOCK_ID_0).unwrap().is_none());
+
+    let block = block_zero();
+    let state_diff = state_diff_zero();
+
+    backend.store_block(block.clone(), state_diff.clone(), vec![]).unwrap();
+
+    assert_eq!(backend.get_block_hash(&BLOCK_ID_0).unwrap().unwrap(), block.info.block_hash().unwrap());
+    assert_eq!(BLOCK_ID_0.resolve_db_block_id(backend).unwrap().unwrap(), BLOCK_ID_0);
+    assert_eq!(backend.get_block_info(&BLOCK_ID_0).unwrap().unwrap(), block.info);
+    assert_eq!(backend.get_block_inner(&BLOCK_ID_0).unwrap().unwrap(), block.inner);
+    assert_eq!(backend.get_block(&BLOCK_ID_0).unwrap().unwrap(), block);
+    assert_eq!(backend.get_block_n(&BLOCK_ID_0).unwrap().unwrap(), 0);
+    assert_eq!(backend.get_block_state_diff(&BLOCK_ID_0).unwrap().unwrap(), state_diff);
+}
+
+#[tokio::test]
+async fn test_store_pending_block() {
+    const BLOCK_ID_PENDING: DbBlockId = DbBlockId::Pending;
+
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    assert!(backend.get_block(&BLOCK_ID_PENDING).unwrap().is_none());
+
+    let block = block_pending_one();
+    let state_diff = state_diff_pending_one();
+
+    backend.store_block(block.clone(), state_diff.clone(), vec![]).unwrap();
+
+    assert!(backend.get_block_hash(&BLOCK_ID_PENDING).unwrap().is_none());
+    assert_eq!(backend.get_block_info(&BLOCK_ID_PENDING).unwrap().unwrap(), block.info);
+    assert_eq!(backend.get_block_inner(&BLOCK_ID_PENDING).unwrap().unwrap(), block.inner);
+    assert_eq!(backend.get_block(&BLOCK_ID_PENDING).unwrap().unwrap(), block);
+    assert_eq!(backend.get_block_state_diff(&BLOCK_ID_PENDING).unwrap().unwrap(), state_diff);
+}
+
+#[tokio::test]
+async fn test_erase_pending_block() {
+    const BLOCK_ID_PENDING: DbBlockId = DbBlockId::Pending;
+
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    backend.store_block(block_zero(), state_diff_zero(), vec![]).unwrap();
+    backend.store_block(block_pending_one(), state_diff_pending_one(), vec![]).unwrap();
+    backend.clear_pending_block().unwrap();
+
+    assert!(backend.get_block(&BLOCK_ID_PENDING).unwrap().is_none());
+
+    backend.store_block(block_one(), state_diff_one(), vec![]).unwrap();
+
+    let block = block_pending_two();
```

Comment:
that's a bit misleading given that it also refers to a pending block for block number one